Sources/Testing/Issues/Issue.swift

```diff
@@ -179,3 +179,217 @@ extension Issue.Kind: CustomStringConvertible {
     }
   }
 }
+
+// MARK: - Codable
+
+extension Issue {
+  /// A serializable type describing a failure or warning which occurred during a test.
+  @_spi(ExperimentalSnapshotting)
+  public struct Snapshot: Sendable, Codable {
+    /// The kind of issue this value represents.
+    public var kind: Kind.Snapshot
+
+    /// Any comments provided by the developer and associated with this issue.
+    ///
+    /// If no comment was supplied when the issue occurred, the value of this
+    /// property is the empty array.
+    public var comments: [Comment]
+
+    /// A ``SourceContext`` indicating where and how this issue occurred.
+    public var sourceContext: SourceContext
+
+    /// Whether or not this issue is known to occur.
+    public var isKnown = false
+
+    /// Initialize an issue instance with the specified details.
+    ///
+    ///
+    init(issue: Issue) {
+      self.kind = Issue.Kind.Snapshot(kind: issue.kind)
+      self.comments = issue.comments
+      self.sourceContext = issue.sourceContext
+      self.isKnown = issue.isKnown
+    }
+  }
+}
+
+extension Issue.Kind {
+  /// Serializable kinds of issues which may be recorded.
+  @_spi(ExperimentalSnapshotting)
+  public enum Snapshot: Sendable, Codable {
+    /// An issue which occurred unconditionally, for example by using
+    /// ``Issue/record(_:fileID:filePath:line:column:)``.
+    case unconditional
+
+    /// An issue due to a failed expectation, such as those produced by
+    /// ``expect(_:_:sourceLocation:)``.
+    ///
+    /// - Parameters:
+    ///   - expectation: The expectation that failed.
+    case expectationFailed(_ expectation: Expectation.Snapshot)
+
+    /// An issue due to a confirmation being confirmed the wrong number of
+    /// times.
+    ///
+    /// - Parameters:
+    ///   - actual: The number of times ``Confirmation/confirm(count:)`` was
+    ///     actually called.
+    ///   - expected: The expected number of times
+    ///     ``Confirmation/confirm(count:)`` should have been called.
+    ///
+    /// This issue can occur when calling
+    /// ``confirmation(_:expectedCount:fileID:filePath:line:column:_:)`` when
+    /// the confirmation passed to these functions' `body` closures is confirmed
+    /// too few or too many times.
+    case confirmationMiscounted(actual: Int, expected: Int)
+
+    /// An issue due to an `Error` being thrown by a test function and caught by
+    /// the testing library.
+    ///
+    /// - Parameters:
```

Comment:
Please ensure documentation matches code. (Since this documentation was copied verbatim from the original type, we can probably slim it down and don't need as much detail. We can refer readers back to the original type for more info.)